### PR TITLE
Restructure the segment structure

### PR DESCRIPTION
**Pull request by cmcuza**
This PR restructures the Segment structure to use the Point structure for its attributes start_point and end_point. All functions using the Segment structure have been updated accordingly as well as documentation.

### CODE REVIEW

File: src/functional/swing_slide_filter.zig [hunk 1]
```diff
@@ -290,10 +278,10 @@ fn appendValue(comptime T: type, value: T, compressed_values: *std.ArrayList(u8)
     }
 }
 
-/// Computes the intercept coefficient of a linear function that passes through the point
-/// (time, value) and with the given slope coefficient.
-fn computeInterceptCoefficient(slope: f80, time: usize, value: f64) f80 {
-    return value - slope * usizeToF80(time);
+/// Computes the intercept coefficient of a linear function that passes through `point` and
+/// with the given slope coefficient.
```
Comment (skejserjensen) [inline src/functional/swing_slide_filter.zig:282]: Very minor, but either with or and. Also slope lacks backticks.

File: src/tersets.zig [hunk 2]
```diff
@@ -39,6 +39,9 @@ pub const Method = enum {
 /// A point represented by `time` and `value`.
 pub const Point = struct { time: usize, value: f64 };
 
+/// Segment represented by the `start_point` and `end_point`.
```
Comment (skejserjensen) [inline src/tersets.zig:42]: Maybe state what a segment is before stating what it is represented by. I do not think it is as self explanatory as point.

**Review by skejserjensen — CHANGES_REQUESTED** [review]

File: src/functional/swing_slide_filter.zig (hunk 1 above)
Comment (cmcuza) [reply src/functional/swing_slide_filter.zig:282]: Fixed.

File: src/tersets.zig (hunk 2 above)
Comment (cmcuza) [reply src/tersets.zig:42]: This is tricky. Finding a proper definition of a Segment without creating confusion is not straightforward. 
Turns out that the proper definition is:  Part of a line that is bounded by two distinct endpoints.
However, that inserts the concept line, which can be problematic. I propose the following:
/// This structure models a straight line segment from `start_point` to `end_point`.

**Review by skejserjensen — APPROVED** [review]

**Pull request merged**